snippets/base/admin/adminmodels.py

```diff
@@ -103,9 +104,337 @@ class AddonAdmin(admin.ModelAdmin):
     list_display = ('name', 'guid')
 
 
+class IconAdmin(admin.ModelAdmin):
+    search_fields = [
+        'name',
+        'image',
+    ]
+    readonly_fields = [
+        'height',
+        'width',
+        'preview',
+        'creator',
+        'created',
+        'snippets',
+    ]
+    list_display_links = [
+        'id',
+        'name',
+    ]
+    list_display = [
+        'id',
+        'name',
+        'width',
+        'height',
+        'preview',
+    ]
+
+    class Media:
+        css = {
+            'all': (
+                'css/admin/IconListSnippets.css',
+            )
+        }
+
+    def save_model(self, request, obj, form, change):
+        if not obj.creator_id:
+            obj.creator = request.user
+        super().save_model(request, obj, form, change)
+
+    def preview(self, obj):
+        text = f'<img style="max-width:120px; max-height:120px;" src="{obj.image.url}"/>'
+        return mark_safe(text)
+
+    def get_readonly_fields(self, request, obj=None):
+        fields = super().get_readonly_fields(request, obj)
+
+        # Don't allow image to be changed after set.
+        if obj and 'image' not in fields:
+            fields.append('image')
+
+        return fields
+
+    def snippets(self, obj):
+        """Snippets using this icon.
+
+        Needs that fancy code b/c icons have multiple relations with multiple Templates.
```

Comment:
Fancy! 🎩

Reply:
fancy, indeed

Reply:
Moved the fancy code in `Icon` model to make it re-usable. ;)